tests/categorical/test_contingency.py

```diff
@@ -0,0 +1,198 @@
+    table = match.make_table(simple_forecast, simple_obs)
+    table2 = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    counts = table.get_counts()
+    actual_table = table.get_table()
+
+    # Test event tables creation matches the stored tables
+    fcst_events, obs_events = match.make_event_tables(simple_forecast, simple_obs)
+    fcst_events2, obs_events2 = match.make_event_tables(simple_forecast, simple_obs, event_threshold=1.3)
+    assert (fcst_events == table.forecast_events).all()
+    assert (fcst_events2 == table.forecast_events).all()
+    assert (obs_events == table.observed_events).all()
+    assert (obs_events2 == table.observed_events).all()
+    assert (table.forecast_events == table2.forecast_events).all()
+    assert (table.observed_events == table2.observed_events).all()
+
+    # Confirm values in the contingency table are correct
+    assert counts["tp_count"] == 9
+    assert counts["tn_count"] == 6
+    assert counts["fp_count"] == 2
+    assert counts["fn_count"] == 1
+    assert counts["total_count"] == 18
+
+    # Confirm that the xarray table object has the correct counts too
+    assert actual_table.sel(contingency="tp_count") == 9
+    assert actual_table.sel(contingency="tn_count") == 6
+    assert actual_table.sel(contingency="fp_count") == 2
+    assert actual_table.sel(contingency="fn_count") == 1
+    assert actual_table.sel(contingency="total_count") == 18
+
+    # Confirm calculations of metrics are correct
+    assert table.accuracy() == (9 + 6) / 18
+    assert table.probability_of_detection() == 9 / (9 + 1)
+    assert table.false_alarm_rate() == 2 / (2 + 6)
+    assert table.threat_score() == 9 / (9 + 2 + 6)
+    assert table.frequency_bias() == (9 + 2) / (9 + 1)
+    assert table.hit_rate() == 9 / (9 + 1)
+    assert table.probability_of_detection() == 9 / (9 + 1)
+    assert table.success_ratio() == 9 / (9 + 2)
+    assert table.specificity() == 6 / (6 + 2)
+
+    # These methods are redirects to eachother
+    assert table.critical_success_index() == table.threat_score()
+    assert table.sensitivity() == table.hit_rate()
+    assert table.false_alarm_rate() == table.probability_of_false_detection()
+
+    pierce_component_a = 9 / (9 + 1)
+    pierce_component_b = 1 / (1 + 6)
+    pierce_expected = pierce_component_a - pierce_component_b
+    assert table.pierce_skill_score() == pierce_expected
+
+
+def test_categorical_table_dims_handling():
+    """
+    Test that the transform function correctly allows dimensional transforms
+    """
+    match = scores.categorical.ThresholdEventOperator()
+    table = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    transformed = table.transform(preserve_dims=["height"])
+
+    acc_withheight = transformed.accuracy()
+    assert acc_withheight.sel(height=10).sum().values.item() == 8 / 9
+    assert acc_withheight.sel(height=20).sum().values.item() == 7 / 9
```

Comment:
Would be nicer to just check if `acc_withheight` equals the expected object. That way we don't need to `.sum()`

Reply:
It's a bit more syntax creating the dataarray, if we're just testing the value of the amount I think it's okay to leave it. Feel free to push in a change if you think it's worthwhile.